### PR TITLE
Backup

### PR DESCRIPTION
**Pull request by amper-fb**

### CODE REVIEW

File: src/AppStore.ts [hunk 1]
```diff
@@ -57,6 +48,7 @@ export const useAppStore = create<IAppState>()((set, get) => {
     loggedUser: authManager.loggedUser,
     userId: null,
     walletId: null,
+    latestBackup: null,
```
Comment (gilamran) [inline src/AppStore.ts:51]: Feels like this should be a component local state... and the `getLatestBackup` should return it to the UI. or keep it on the store but rename it to `loadLatestBackup`

File: src/AppStore.ts [hunk 2]
```diff
@@ -145,42 +142,54 @@ export const useAppStore = create<IAppState>()((set, get) => {
...
-      rememberBackupPassphrase(passphrase, userId);
-      set((state) => ({ ...state, passphrase }));
+      const latestBackup = await apiService.getLatestBackup(walletId);
+      set((state) => ({ ...state, latestBackup }));
     },
-    regeneratePassphrase: () => {
-      const { userId } = get();
-      if (!userId) {
-        throw new Error("First login to demo app server");
+    getPassphraseInfos: async () => {
```
Comment (gilamran) [inline src/AppStore.ts:156]:
```suggestion
    loadPassphraseInfos: async () => {
```

File: src/components/BackupAndRecover.tsx [hunk 3]
```diff
@@ -54,46 +197,43 @@ export const BackupAndRecover: React.FC = () => {
...
-    action: doRecoverKeys,
-    isDisabled: isRecoverInProgress || passphrase === null || passphrase.trim() === "",
+  const appleBackupAction: ICardAction = {
+    label: "iCloud Backup",
+    action: () => doBackupKeys(() => passphrasePersist(PassphraseLocation.iCloud)),
+    isDisabled: !appleSignedIn || isRecoverInProgress || isBackupInProgress || hasReadyAlgo === false,
+    isInProgress: isBackupInProgress,
+  };
+
+  const RecoverAction: ICardAction = {
```
Comment (gilamran) [inline src/components/BackupAndRecover.tsx:216]:
```suggestion
  const recoverAction: ICardAction = {
```

File: src/services/GoogleDrive.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import { DISCOVERY_DOC } from "../auth/providers";
+
+export const gdriveRecover = async (token: string, passphraseId: string) => {
```
Comment (gilamran) [inline src/services/GoogleDrive.ts:3]:
```suggestion
export const gdriveRecover = (token: string, passphraseId: string) => {
```

File: src/services/GoogleDrive.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        }
+      },
+      onerror: reject,
+      ontimeout: reject,
+      timeout: 5_000,
+    });
+  });
+};
+
+export const gdriveBackup = async (token: string, passphrase: string, passphraseId: string) => {
```
Comment (gilamran) [inline src/services/GoogleDrive.ts:42]:
```suggestion
export const gdriveBackup = (token: string, passphrase: string, passphraseId: string) => {
```

File: src/components/BackupAndRecover.tsx [hunk 6]
```diff
@@ -1,24 +1,166 @@
-import React from "react";
+import React, { useEffect } from "react";
```
Comment (gilamran) [inline src/components/BackupAndRecover.tsx:1]: Let's talk about this file, it feel too complicated... can't say that I fully understand what's going on.

File: index.html [hunk 7]
```diff
@@ -5,6 +5,8 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>Fireblocks Js cosigner</title>
     <link href="css/index.css" rel="stylesheet">
+    <script src="https://apis.google.com/js/api.js"></script>
+    <script src="https://cdn.apple-cloudkit.com/ck/2/cloudkit.js"></script>
```
[the remark this answers is not in the document]
Comment (amper-fb) [reply index.html:9]: nope

File: package.json [hunk 8]
```diff
@@ -38,6 +42,7 @@
     "postcss": "^8.4.23",
     "prettier": "^3.0.0",
     "tailwindcss": "^3.3.2",
+    "tsl-apple-cloudkit": "^0.2.31",
```
[the remark this answers is not in the document]
Comment (amper-fb) [reply package.json:45]: what?

File: src/services/GoogleDrive.ts (hunk 4 above)
Comment (amper-fb) [reply src/services/GoogleDrive.ts:3]: why? it returns a promise, therefore async

File: src/services/GoogleDrive.ts (hunk 5 above)
Comment (amper-fb) [reply src/services/GoogleDrive.ts:42]: same as above.

File: package.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (amper-fb) [reply package.json:45]: oh, I understand, the index.html file adds the script, here we just use the types, see how I import it, and the fact that it's devdep + excluded from bundle

File: .env [hunk 9]
```diff
@@ -1,9 +1,11 @@
 # VITE_AUTOMATE_INITIALIZATION=true
 
 # SANDBOX
-VITE_BACKEND_BASE_URL=https://ncw-demo-dev.2uaqu5aka49io.eu-central-1.cs.amazonlightsail.com
-VITE_NCW_SDK_ENV=sandbox
+# VITE_BACKEND_BASE_URL=https://ncw-demo-dev.2uaqu5aka49io.eu-central-1.cs.amazonlightsail.com
```
[the remark this answers is not in the document]
Comment (amper-fb) [reply .env:4]: yes

File: src/components/Cloudkit.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+        containers: [
+          {
+            containerIdentifier: ENV_CONFIG.CLOUDKIT_CONTAINER_ID,
+            apiTokenAuth: {
+              apiToken: ENV_CONFIG.CLOUDKIT_APITOKEN,
+              persist: true,
+              signInButton: { id: "sign-in-button", theme: "black" },
+              signOutButton: { id: "sign-out-button", theme: "black" },
+            },
+            environment: "development",
```
[the remark this answers is not in the document]
Comment (amper-fb) [reply src/components/Cloudkit.ts:21]: if someone takes this to production they can change it, for our demo it will probably be kept as development

**Pull request merged**